amp/WEB-INF/src/org/digijava/module/aim/dbentity/ApprovalStatusType.java

```diff
@@ -0,0 +1,77 @@
+package org.digijava.module.aim.dbentity;
+
+import java.io.Serializable;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import org.hibernate.HibernateException;
+import org.hibernate.engine.spi.SessionImplementor;
+import org.hibernate.type.StringType;
+import org.hibernate.usertype.UserType;
+
+/**
+ * @author Octavian Ciubotaru
+ */
+public class ApprovalStatusType implements UserType {
```

Comment:
I think you can do the same by using 'org.hibernate.type.EnumType' in .hbm.xml file (e.g. https://github.com/devgateway/amp/blob/a93e1fe146b66eb2c2933968abd9d694759b626c/amp/WEB-INF/src/org/digijava/module/aim/dbentity/AmpMenuEntry.hbm.xml#L63) and configure "useNamed" to true to keep strings.

Reply:
Using EnumType will require changing `startedapproved` to `STARTED_APPROVED_STATUS` in database. I wanted to avoid renaming to prevent possible bugs coming out of this.

Reply:
doesn't it get the .name that you can configure to return the right value?

Reply:
Could you please elaborate more on that? From what I see there is no way to do this.

Reply:
My bad memories about code level only hacks - not applicable in this case. Though it would be nice to avoid this class and may be indeed use a patch to convert DB entries?

Btw I saw org.hibernate.type.EnumType already implements UserType so could be simpler to extend it (with `nullSafeGet` and `nullSafeSet` only)?

Reply:
I was going to suggest the same, to override EnumType .that we can also do it with AmpOfflineVersionType that follows the same logic, and we could have smaller classes

Reply:
Re-implemented enum as a subclass of EnumType. AmpOfflineVersionType was left untouched because AmpOfflineVersion is not an enum.

Reply:
right my bad, I overlooked it